src/lib/index/index_cp.hpp

```diff
@@ -77,13 +70,91 @@ struct IndexCPContext : public CPContext {
     BlkId* next_blkid() { return m_free_node_blkid_list->next(m_buf_it.free_node_list_it); }
     std::string to_string() const {
         std::string str{
-            fmt::format("IndexCPContext cpid={} dirty_buf_count={} dirty_buf_list_size={} blkid_list_size={}", id(),
+            fmt::format("IndexCPContext cpid={} dirty_buf_count={} dirty_buf_list_size={} blkid_list_size={}\n", id(),
                         m_dirty_buf_count.get(), m_dirty_buf_list->size(), m_free_node_blkid_list->size())};
 
-        // TODO dump all index buffers.
+        // Mapping from a node to all its parents in the graph.
+        // Display all buffers and its dependencies and state.
+        std::unordered_map< IndexBuffer*, std::vector< IndexBuffer* > > parents;
+        IndexBufferPtr* buf;
```

Comment:
show a buf and dependencies
ex:
IndexBuffer 0x60b000055a80 blkid=562954248388608 state=1 node_buf=0x6210000a3e00 next_buffer=0x60b000055870 wait_for=2
Depends: 0x60b00004bdc0(1) 0x60b000055b30(1)